File: matacc.c
#include <stdio.h>

int main(int argc, char *argv[]){
    if(argc != 4){
        fprintf(stderr,
                "Usage: time %s N time type\n"
                "\n"
                "type: 0 - ij loop\n"
                "      1 - ji loop\n"
                "      2 - one row\n"
                "      3 - one col\n"
                "      4 - diag\n"
                , argv[0]);
        return -1;
    }
    int N = atoi(argv[1]);
    int T = atoi(argv[2]);
    char a[N][N];
    int i, j;
    while(--T){
        switch(argv[3][0]){
        case '0':
            for(i = 0; i < N; i++)
                for(j = 0; j < N; j++)
                    a[i][j] = 42;
            break;
        case '1':
            for(j = 0; j < N; j++)
                for(i = 0; i < N; i++)
                    a[i][j] = 42;
            break;
        case '2':
            for(i = 0; i < N; i++)
                a[0][i] = 42;
            break;
        case '3':
            for(i = 0; i < N; i++)
                a[i][0] = 42;
            break;
        case '4':
            for(i = 0; i < N; i++)
                a[i][i] = 42;
            break;
        }
    }
    return 0;
}

File: matacc.py
#!/usr/bin/python

from sys import exit, argv, stderr

def ij_loop(m):
    N = len(m)
    for i in xrange(N):
        for j in xrange(N):
            m[i][j] = 42

def ji_loop(m):
    N = len(m)
    for j in xrange(N):
        for i in xrange(N):
            m[i][j] = 42

def one_row(m):
    N = len(m)
    for i in xrange(N):
        m[0][i] = 42

def one_col(m):
    N = len(m)
    for i in xrange(N):
        m[i][0] = 42

def diag(m):
    N = len(m)
    for i in xrange(N):
        m[i][i] = 42

def main():
    if len(argv) != 4:
        stderr.write('''Usage: time %s N time type

    type: 0 - ij loop
          1 - ji loop
          2 - one row
          3 - one col
          4 - diag
''' % argv[0])
        exit(-1)


    N = int(argv[1])
    T = int(argv[2])
    m = [[0 for x in xrange(N)] for x in xrange(N)] 

    for i in xrange(T):
        [ij_loop, ji_loop, one_row, one_col, diag][int(argv[3])](m)

    exit(0)

if __name__ == "__main__":
    main()
